src/application/services/useNote.ts

```diff
@@ -166,12 +167,20 @@ export default function (options: UseNoteComposableOptions): UseNoteComposableSt
     /**
      * @todo try-catch domain errors
      */
-    const response = await noteService.getNoteById(id);
-
-    note.value = response.note;
-    canEdit.value = response.accessRights.canEdit;
-    noteTools.value = response.tools;
-    parentNote.value = response.parentNote;
+    try {
+      const response = await noteService.getNoteById(id);
+
+      note.value = response.note;
+      canEdit.value = response.accessRights.canEdit;
+      noteTools.value = response.tools;
+      parentNote.value = response.parentNote;
+    } catch (error) {
+      if (error instanceof NotFoundError) {
+        void router.push(`/error/404?message=${error.message}`);
```

Comment:
It's better to specify error message directly, instead of using `error.message`. And use i18n for that.

Other code looks good for me